### PR TITLE
Update review_decision and tests using old enum types

### PR DESCRIPTION
**Pull request by nmadruga**
Just a tiny change that was on my TODO list

I haven't checked with all tests... But I think many still need to be updated after latest changed to the test data.

### CODE REVIEW

File: database/insertData/06_applications_D_reviewTest.js [hunk 1]
```diff
@@ -572,4 +572,118 @@ exports.queries = [
...
+                templateElementId: 4008
+                status: SUBMITTED 
+                value: { text: "Amoxicilin" }
+              }
+              {
+                id: 4156
+                isValid: true
+                templateElementId: 4009
+                status: SUBMITTED 
+                value: { text: "Medicine", optionIndex: 2 }
```
Comment (CarlosNZ) [inline database/insertData/06_applications_D_reviewTest.js:660]: Dropdown responses need to be in the form
```
{"text": "Medicine", "selection": "Medicine", "optionIndex": 0}
```
Also -- in the template "Medicine" is Option Index 0 (first item)

File: plugins/action_update_review_status/src/databaseMethods.ts [hunk 2]
```diff
@@ -9,13 +9,39 @@ const databaseMethods = (DBConnect: any) => ({
...
-      ON review.review_assignment_id = review_assignment.id
-      JOIN review_status_history ON review.id = review_status_history.review_id
-      WHERE review_status_history.is_current = true
-      AND review_assignment.application_id = $1
-      AND stage_id = $2
-      AND review_assignment.level_number = $3
+      FROM review_assignment 
+        LEFT JOIN review ON review_assignment.id = review.review_assignment_id 
+        LEFT JOIN review_status_history ON review.id = review_status_history.review_id 
+          AND review_status_history.is_current = true
```
Comment (CarlosNZ) [inline plugins/action_update_review_status/src/databaseMethods.ts:15]: Just wondering -- we currently have created Views for application_stage and status history (`application_stage_status_all` and `application_stage_status_latest`) which has been quite useful. Do you think we should have one for review histories too?

File: plugins/action_update_review_status/src/updateReviewsStatuses.ts [hunk 3]
```diff
@@ -99,12 +121,16 @@ const updateReviewsStatuses: ActionPluginType = async ({
...
+
+      // Get reviews matching application_id, current stage, in level 1
       const level = 1
-      // Get reviews/review assignments (with status) matching application_id & current stage
       const reviews = await getReviewsByLevelAndStatus(level)
       // Deduce which ones should be updated to Pending
       for (const review of reviews) {
         const { reviewAssignmentId, reviewStatus } = review
+        if (reviewStatus === ReviewStatus.Locked) {
+          // TODO: Check with team should also lock the reviewAssignment ?
```
Comment (CarlosNZ) [inline plugins/action_update_review_status/src/updateReviewsStatuses.ts:132]: Did you get an answer on this, and do you want to figure it out before merging this PR, or wait till later?

File: src/components/postgresConnect.ts [hunk 4]
```diff
@@ -551,6 +551,18 @@ class PostgresDB {
...
+      'INSERT into review_assignment is_locked VALUES ($2) WHERE id = $1 RETURNING id, is_locked'
+    try {
+      const result = await this.query({ text, values: [reviewAssignmnetId, isLocked] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
```
Comment (CarlosNZ) [inline src/components/postgresConnect.ts:565]: Should probably keep this in the databaseMethods file of the plugin for consistency (unless this is used/intended to be used elsewhere?)

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Looks great, and tests all pass. Just a couple of fairly minor comments.

File: src/components/postgresConnect.ts (hunk 4 above)
Comment (CarlosNZ) [reply src/components/postgresConnect.ts:565]: Never mind, I see you already did this in your next PR :)

File: database/insertData/06_applications_D_reviewTest.js (hunk 1 above)
Comment (nmadruga) [reply database/insertData/06_applications_D_reviewTest.js:660]: Oops! Good point, I did that and end up not testing.

Comment (nmadruga) [reply database/insertData/06_applications_D_reviewTest.js]: Done: 3c3ef77

File: plugins/action_update_review_status/src/databaseMethods.ts (hunk 2 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/databaseMethods.ts:15]: Hmm perhaps. Do you think I could include as part of this PR?

File: plugins/action_update_review_status/src/updateReviewsStatuses.ts (hunk 3 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/updateReviewsStatuses.ts:132]: No... I'll leave this one there until we decide. Will add a discussion issue to check that.

File: plugins/action_update_review_status/src/databaseMethods.ts (hunk 2 above)
Comment (CarlosNZ) [reply plugins/action_update_review_status/src/databaseMethods.ts:15]: If it's not too much extra work. But can make a new issue if you'd prefer.

File: plugins/action_update_review_status/src/updateReviewsStatuses.ts (hunk 3 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/updateReviewsStatuses.ts:132]: Issue #757

File: plugins/action_update_review_status/src/databaseMethods.ts (hunk 2 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/databaseMethods.ts:15]: Hmm I was checking out file `32_review_status_history` and saw that the latest `status` is already exposed in the `review` table:
https://github.com/openmsupply/application-manager-server/blob/9da7a3cb43932aac717af2267430b56387d19039/database/buildSchema/32_review_status_history.sql#L42-L52
So actually we don't need this view... I can just change this code to use review.status instead.

Comment (CarlosNZ) [reply plugins/action_update_review_status/src/databaseMethods.ts:15]: Looks like "status" is visible on `review` table in GraphQL, but not in SQL, as it hasn't been added as a generated column.

Comment (nmadruga) [reply plugins/action_update_review_status/src/databaseMethods.ts:15]: Yes, tried changing and realised that too. So will just leave this one as is.

**Review by CarlosNZ — APPROVED** [review]

**Pull request merged**